app/editor/components/SelectionToolbar.tsx

```diff
@@ -229,6 +232,12 @@ export default class SelectionToolbar extends React.Component<Props> {
       return null;
     }
 
+    // Only "code_inline" option should be visible if
+    // it's an inline code selection.
+    if (isCodeInlineSelection) {
+      items.map((item) => (item.visible = item.name === "code_inline"));
```

Comment:
Sorry, maybe I could have been clearer – `getFormattingMenuItems` should return the correct menu items given the state, we don't want to have additional processing here as well. The `visible` prop should be modified within that method

Reply:
You definitely did mentioned what you were looking for

> **all of these** [items] would have **visible=false when inside code**

so it's totally my bad. Sorry.

Also

>  actually a query for this, `isInCode(state)`

and 

https://github.com/outline/outline/blob/4c05fe422c73ebd18526c37c74fce8bbdf726f1c/app/editor/menus/formatting.ts#L29-L30

should have been good hints

Third times the charm!